theano/sandbox/cuda/__init__.py

```diff
@@ -452,8 +592,28 @@ def use(device,
 
             if config.print_active_device:
                 cnmem_enabled = "enabled" if config.lib.cnmem else "disabled"
-                print("Using gpu device %d: %s (CNMeM is %s)" % (
-                        active_device_number(), active_device_name(), cnmem_enabled), file=sys.stderr)
+                cudnn_version = "not available"
+                warn = None
+                try:
+                    (hdr_v, runtime_v) = dnn_version()
+                    cudnn_version = runtime_v
+                    # 4100 should not print warning with cudnn 4 final.
+                    if cudnn_version > 4100:
+                        warn = ("Your CuDNN version is more recent then Theano."
+                                " If you see problems, try updating Theano or"
+                                " downgrading CuDNN to version 4.")
+                except Exception, e:
+                    pass
+                print("Using gpu device %d: %s (CNMeM is %s, CuDNN %s)" % (
+                    active_device_number(),
+                    active_device_name(),
+                    cnmem_enabled,
+                    cudnn_version,),
+                      file=sys.stderr)
```

Comment:
I think it would be useful to mention how much memory CNMeM is configured to claim, at least when it's enabled by default (it still isn't, right?). Otherwise people will wonder why their simple model already takes so much GPU memory in Theano compared to other libraries. Something like:
`Using gpu device 0: Geforce GT 640 (CNMeM enabled: 45% of memory; CuDNN 4004)`
(If not in this PR, I can file an issue.)